vermouth/processors/do_links.py

```diff
@@ -277,13 +277,17 @@ def _build_link_interaction_from(molecule, interaction, match):
 class DoLinks(Processor):
     def run_molecule(self, molecule):
         links = molecule.force_field.links
+        _nodes_to_remove = []
         for link in links:
             matches = match_link(molecule, link)
             for match in matches:
                 for node, node_attrs in link.nodes.items():
                     if 'replace' in node_attrs:
-                        node_mol = molecule.nodes[match[node]]
-                        node_mol.update(node_attrs['replace'])
+                        if node_attrs['replace']['atomname'] is None:
```

Comment:
This will break if `'atomname'` is not in `node_attrs['replace']`. Replace with `if node_attrs['replace'].get('atomname', False):`